07-http_gordle/final/internal/handlers/getstatus/getstatus.go

```diff
@@ -0,0 +1,50 @@
+package getstatus
+
+import (
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+
+	"learngo-pockets/httpgordle/api"
+	"learngo-pockets/httpgordle/internal/domain"
+	"learngo-pockets/httpgordle/internal/handlers/apiconversion"
+	"learngo-pockets/httpgordle/internal/repository"
+)
+
+type gameFinder interface {
+	Find(domain.GameID) (domain.Game, error)
+}
+
+// Handler returns the handler for the game finder endpoint.
+func Handler(repo gameFinder) http.HandlerFunc {
```

Comment:
Just for information... no need to change. One way to avoid needing the chi router, is to use a wrapper handler that creates a temporary handler:

```
type Sessions interface {
	ByGameID(GameID) (Game, error)
}

type GamesServer struct {
	sessions Sessions
}

func (server *GamesServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	p := r.Path
	p = strings.TrimPrefix(p, "/")
	gameid, rest, ok := strings.Cut(p, path.)
	if !ok {
		http.Error(... not found ...)
		return
	}

	id, err := strconv.Itoa(gameid)
	if err != nil {
		http.Error(... bad request ...)
		return
	}

	sr := r.Clone()
	sr.Path = rest
	GameServer{
		ID: id,
		sessions: server.sessions,
	}.ServeHTTP(w, sr)
}

type GameServer struct {
	GameID  GameID
	sessions Sessions
}

func (server *GameServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	game, err := server.session.ByGameID(server.GameID)
	if err != nil {
		http.Error(... not found ...)
		return
	}

	...
}
```

The other, more usual, is to use: